webapp/src/redux/features/message-relay/signalRMiddleware.ts

```diff
@@ -146,34 +144,29 @@ export const signalRMiddleware = (store: StoreMiddlewareAPI) => {
 };
 
 export const registerSignalREvents = (store: Store) => {
-    hubConnection.on(SignalRCallbackMethods.ReceiveMessage, (message: IChatMessage, chatId: string) => {
-        store.dispatch({ type: 'conversations/updateConversationFromServer', payload: { message, chatId } });
-    });
-
-    hubConnection.on(SignalRCallbackMethods.ReceiveResponse, (askResult: IAskResult, chatId: string) => {
-        const loggedInUserId = store.getState().app.activeUserInfo?.id;
-        const originalMessageUserId: string | undefined = askResult.variables.find((v) => v.key === 'userId')?.value;
-        const isPlanForLoggedInUser = loggedInUserId === originalMessageUserId;
-        const messageType = Number(askResult.variables.find((v) => v.key === 'messageType')?.value) as ChatMessageType;
-
-        const message = {
-            type: messageType,
-            timestamp: new Date().getTime(),
-            userName: 'bot',
-            userId: 'bot',
-            content: askResult.value,
-            prompt: askResult.variables.find((v) => v.key === 'prompt')?.value,
-            authorRole: AuthorRoles.Bot,
-            id: askResult.variables.find((v) => v.key === 'messageId')?.value,
-            state:
-                messageType === ChatMessageType.Plan && isPlanForLoggedInUser
+    hubConnection.on(SignalRCallbackMethods.ReceiveMessage, (chatId: string, userId: string, message: IChatMessage) => {
+        if (message.authorRole === AuthorRoles.Bot) {
+            const loggedInUserId = store.getState().app.activeUserInfo?.id;
+            const responseToLoggedInUser = loggedInUserId === userId;
+            message.planState =
+                message.type === ChatMessageType.Plan && responseToLoggedInUser
                     ? PlanState.PlanApprovalRequired
-                    : PlanState.Disabled,
-        } as IChatMessage;
+                    : PlanState.Disabled;
+        }
 
-        store.dispatch({ type: 'conversations/updateConversationFromServer', payload: { message, chatId } });
+        store.dispatch({ type: 'conversations/addMessageToConversationFromServer', payload: { chatId, message } });
     });
 
+    hubConnection.on(
+        SignalRCallbackMethods.ReceiveMessageContent,
```

Comment:
nit: use `ReceiveMessageStream` instead of `ReceiveMessageContent` to be more indicative of content coming through, MessageContent and Message can be mistaken as synonymous